packages/eas-cli/src/project/publish.ts

```diff
@@ -16,6 +16,10 @@ import { expoCommandAsync } from '../utils/expoCli';
 import uniqBy from '../utils/expodash/uniqBy';
 
 export const TIMEOUT_LIMIT = 60_000; // 1 minute
+export const MAX_ASSETS_PER_UPLOAD = 600; // https://github.com/expo/universe/blob/main/server/www/src/graphql/mutations/AssetMutation.ts#L14
```

Comment:
just copy/pasted the # found in `www`. we'll have to remember to keep these two numbers in sync... not sure there's a better way to handle this?

Reply:
how would you feel about adding some field to the gql return return value for `PublishMutation.getUploadURLsAsync` or `PublishQuery.getAssetMetadataAsync`? I don't feel like either of those queries really match up with returning `MAX_ASSETS_PER_UPLOAD`, but it feels better than adding an additional endpoint & request just for this data.

Reply:
I think this is something to discuss with @wschurman. Yesterday he started optimizing the update-publishing flow, which is bottlenecked on the number of assets and has several checks for the asset count.

I could see two approaches that are mutually compatible: expose the asset limits through a GraphQL API, or allow existing GraphQL APIs to return an error message or warning message that is EAS CLI is then responsible for displaying.

In either case, you may want to back out these changes from this PR and move them to another PR since everything else here looks ready to land.